### ISSUE TITLE
Why is everything observable?

### ISSUE THREAD

**Issue by Xerus2000** [dlemmermann/PreferencesFX#17]
While inspecting the code I find many Properties whose value is actually immutable, and set only once on object creation. Why are these Properties instead of simple variables?

**Comment by martinfrancois** [comment]
Hi @Xerus2000 
Thanks for your comment! We tried to use observables wherever it makes sense. Could you please elaborate on which properties you mean exactly? I will then be able to tell you the reasons (if there are) and else we could discuss about changing them into simple variables, should it not make any sense.

**Comment by Xerus2000** [comment]
For example a `Change` in the `History` should not be mutable and thus have no Properties. Of you ever use Kotlin you'll suddenly start adopting and appreciating immutability much more, I strongly suggest giving it a try ;)
Furthermore, I don't see a reason why a Change would have separate properties for Lists and Single Items. But I'm already working on a PR that addresses that.

**Comment by martinfrancois** [comment]
The `Change` is immutable for the most part, you will notice that the only things that can be changed externally are the old and the new list / values.  The rest is only exposed as `ReadOnly` properties.

Actually, making them mutable in this regard was a tradeoff. The problem were the compounded changes - let's imagine you drag a slider from 1 to 42. As you do so, JavaFX will trigger hundreds if not thousands of change events. If `Change` were actually immutable, we would create hundreds of thousands of change objects during that which would become obsolete in microseconds. This is very wasteful and causes unnecessary garbage collection and pollution of the heap. This is why we decided to set the new value of the previous change in case of a compounded change to the changed value, to only have one object and that works very well.
Trying out Kotlin is also on my list, maybe I will appreciate them more then ;). I already learned to appreciate them in Haskell, but in this situation I found this to be the better tradeoff.

Considering the thing with lists and single items... If you manage to make it work without, I'll be very happy to merge it. We tried coming up with a solution that would work without it but didn't succeed. The problem here is that it could be that in one change there are a list of changes that happen (for example, if in the demo you select or deselect multiple favorite apps at one time). Saving every regular change in a list without mirroring that onto a regular ObjectProperty caused much more hassle than it was worth.
